### PR TITLE
fixing moderator overview on reshares

### PR DESCRIPTION
**Pull request by ArtificialOwl**
- fix a call to a deprecated method is returns a non managed exception
- it now uses CircleManager. Still needs to check its availability as there is no integration of circles within core. left a TODO note.
- This is the most efficient way to know if a user is a (inherited) member of a circle and his access level

### CODE REVIEW

**Comment by ArtificialOwl** [comment]
(sorry for the mess, experience edge experience, and UI had glitches :])

**Comment by CarlSchwan** [comment]
> Psalm is not happy :)

To fix it, run:

```
composer install
composer psalm:update-baseline
```

From that I could see, it's just that we have less warning now :)

**Comment by ArtificialOwl** [comment]
@CarlSchwan @skjnldsv new errors does not seems related !

**Comment by ArtificialOwl** [comment]
@CarlSchwan @PVince81 @come-nc  can I have some review on this one pleas ? the drone issue does not seems related to current code

File: apps/files_sharing/lib/Controller/ShareAPIController.php [hunk 1]
```diff
@@ -1234,17 +1233,17 @@ public function pendingShares(): DataResponse {
...
+					$node = $nodes[0];
 				}
-			} else {
-				$node = $nodes[0];
-			}
 
-			try {
-				$formattedShare = $this->formatShare($share, $node);
+				try {
+					$formattedShare = $this->formatShare($share, $node);
```
Comment (come-nc) [inline apps/files_sharing/lib/Controller/ShareAPIController.php:1246]: indentation is messed up here

File: apps/files_sharing/lib/Controller/ShareAPIController.php [hunk 2]
```diff
@@ -1745,12 +1744,22 @@ private function shareProviderResharingRights(string $userId, IShare $share, $no
...
+
+				// We get the federatedUser linked to the userId (local user, so type=1)
+				// We browse the federatedUser's membership to confirm it exists and level is moderator
+				$federatedUser = $circleManager->getFederatedUser($userId, 1);
+				foreach($federatedUser->getMemberships() as $membership) {
+					if ($membership->getCircleId() !== $sharedWith) {
+						continue;
+					}
+
+					return ($membership->getLevel() >= 4);
```
Comment (come-nc) [inline apps/files_sharing/lib/Controller/ShareAPIController.php:1760]:
```suggestion
					if ($membership->getCircleId() === $sharedWith) {
						return ($membership->getLevel() >= 4);
					}
```

File: apps/files_sharing/lib/Controller/ShareAPIController.php [hunk 3]
```diff
@@ -1745,12 +1744,22 @@ private function shareProviderResharingRights(string $userId, IShare $share, $no
 				$sharedWith = substr($share->getSharedWith(), $shareWithStart, $shareWithLength);
 			}
 			try {
-				$member = \OCA\Circles\Api\v1\Circles::getMember($sharedWith, $userId, 1);
-				if ($member->getLevel() >= 4) {
-					return true;
+				// TODO: switch to ICirclesManager once we have it available within core
```
Comment (come-nc) [inline apps/files_sharing/lib/Controller/ShareAPIController.php:1747]: When will that be? Is it another PR?

Comment (ArtificialOwl) [reply apps/files_sharing/lib/Controller/ShareAPIController.php:1747]: yes, the implementation of `ICirclesManager` within core needs to be plan for a next major release of NC

File: apps/files_sharing/lib/Controller/ShareAPIController.php (hunk 2 above)
Comment (ArtificialOwl) [reply apps/files_sharing/lib/Controller/ShareAPIController.php:1760]: Ahah <3

**Comment by ArtificialOwl** [comment]
squashed and rebased, fixing @come-nc's comments

File: apps/files_sharing/lib/Controller/ShareAPIController.php (hunk 1 above)
Comment (come-nc) [reply apps/files_sharing/lib/Controller/ShareAPIController.php]: Nope, indentation of the if above is still wrong.
Line 1236 to 1243 have one extra tab that should be removed.
It surprises me that the php-cs in the CI does not check this.

Comment (ArtificialOwl) [reply apps/files_sharing/lib/Controller/ShareAPIController.php]: fixed <3

File: apps/files_sharing/lib/Controller/ShareAPIController.php [hunk 4]
```diff
@@ -1745,12 +1744,20 @@ private function shareProviderResharingRights(string $userId, IShare $share, $no
...
+				// We browse the federatedUser's membership to confirm it exists and level is moderator
+				$federatedUser = $circleManager->getFederatedUser($userId, 1);
+				foreach($federatedUser->getMemberships() as $membership) {
+					if ($membership->getCircleId() === $sharedWith) {
+						return ($membership->getLevel() >= 4);
+					}
 				}
-				return false;
-			} catch (QueryException $e) {
+			} catch (\Exception $e) {
```
Comment (come-nc) [inline apps/files_sharing/lib/Controller/ShareAPIController.php:1760]: This may catch a lot of things, log the exception maybe?

**Review by come-nc — APPROVED** [review]
I cannot vouch for the logic as I do not understand what this does exactly but code looks good now.

File: apps/files_sharing/lib/Controller/ShareAPIController.php [hunk 5]
```diff
@@ -1745,12 +1750,29 @@ private function shareProviderResharingRights(string $userId, IShare $share, $no
...
+				$this->logger->log(0,
+								   'exception while confirming resharing rights visibility',
+								   [
+									   'userId' => $userId,
+									   'sharedWith' => $sharedWith,
+									   'nodeId' => $node->getId()
+								   ]
+				);
+
```
Comment (come-nc) [inline apps/files_sharing/lib/Controller/ShareAPIController.php:1775]:
```suggestion
				$this->logger->info(
								   'Exception while confirming resharing rights visibility',
								   [
									   'userId' => $userId,
									   'sharedWith' => $sharedWith,
									   'nodeId' => $node->getId(),
									   'exception' => $e,
								   ]
				);

```

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by PVince81** [comment]
please check the tests:
```
1) OCA\Files_Sharing\Tests\Controller\ShareAPIControllerTest::testDeleteShareShareNotFound
TypeError: Argument 8 passed to OCA\Files_Sharing\Controller\ShareAPIController::__construct() must implement interface Psr\Log\LoggerInterface, string given, called in /home/runner/work/server/server/apps/files_sharing/tests/Controller/ShareAPIControllerTest.php on line 164
```

**Pull request closed without merging**